code/modules/modular_computers/file_system/programs/messenger/messenger_program.dm

```diff
@@ -608,9 +608,24 @@
 
 	// Show it to ghosts
 	var/ghost_message = span_game_say("[span_name("[sender]")] [rigged ? "(as [span_name(fake_name)]) Rigged " : ""]PDA Message --> [span_name("[signal.format_target()]")]: \"[signal.format_message()]\"")
-	var/list/ghosts = GLOB.dead_player_list + GLOB.current_observers_list
-	for(var/mob/player_mob as anything in ghosts)
-		if(isobserver(player_mob) && (get_chat_toggles(player_mob.client) & CHAT_GHOSTPDA))
+	var/list/message_listeners = list(
+		"dead players list" = GLOB.dead_player_list,
+		"observing players list" = GLOB.current_observers_list,
+	)
```

Comment:
This seems over-engineered for what is going on here

Reply:
Doing it this way to identify null clients in the list w/o having to say null in one of these two lists which is not helpful